kornia/augmentation/_2d/intensity/random_channel_dropout.py

```diff
@@ -0,0 +1,94 @@
+
+import torch
+
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.core import Tensor
+from kornia.core.check import KORNIA_CHECK
+
+
+class RandomChannelDropout(IntensityAugmentationBase2D):
+    """Randomly Drop Channels in the input Image.
+    Args:
+        channel_drop_range: range from which we choose the number of channels to drop.
+        fill_value (int, float): pixel value for the dropped channel.
+        same_on_batch: apply the same transformation across the batch.
+        p: probability of applying the transformation.
+        keepdim: whether to keep the output shape the same as input (True) or broadcast it
+                 to the batch form (False).
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`
+        - Output: :math:`(B, C, H, W)` or :math:`(C, H, W)`
+    Examples:
+        >>> _ =torch.random.manual_seed(1)
+        >>> img = torch.ones(1, 3, 5, 5)
+        >>> out = RandomChannelDropout(p=1)(img)
+        >>> out.shape
+        torch.Size([1, 3, 5, 5])
+        >>> out
+        tensor([[[[0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0.],
+                  [0., 0., 0., 0., 0.]],
+        <BLANKLINE>
+                 [[1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.]],
+        <BLANKLINE>
+                 [[1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.],
+                  [1., 1., 1., 1., 1.]]]])
+    """
+
+    def __init__(
+        self,
+        channel_drop_range: Tuple[int, int] = (1, 1),
+        fill_value: Union[int, float] = 0,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim)
+        self.fill_value = fill_value
+        KORNIA_CHECK(
+            isinstance(channel_drop_range, tuple) and len(channel_drop_range) == 2,
+            "Invalid channel_drop_range. Should be tuple of length 2.",
+        )
+        KORNIA_CHECK(
+            1 <= channel_drop_range[0] <= channel_drop_range[1],
+            "Invalid channel_drop_range. Max channel should be greater than lower.",
+        )
+        self.min_channel = channel_drop_range[0]
+        self.max_channel = channel_drop_range[1]
+
+    def generate_parameters(self, batch_shape: Tuple[int, ...]) -> Dict[str, Tensor]:
+        params = super().generate_parameters(batch_shape)
+        # +1 to avoid possible error, when low = high
+        params["channel_params"] = torch.randint(low=batch_shape[0], high=batch_shape[1] + 1, size=(batch_shape[2],))
+
+        return params
```

Comment:
why not using
```suggestion
    self._param_generator =  rg.PlainUniformGenerator((self. channel_drop_range, "channel_params", None, None))
```

Or implement your own like:
https://github.com/kornia/kornia/blob/master/kornia/augmentation/random_generator/_2d/gaussian_blur.py

Reply:
The problem is to make samples of channels to drop,, based on the number of channels to drop

Reply:
We passed the whole batch_shape to random_generator. You have it.